libbeat/internal/testutil/util.go

```diff
@@ -39,3 +43,35 @@ func SeedPRNG(t *testing.T) {
 	t.Logf("reproduce test with `go test ... -seed %v`", seed)
 	rand.Seed(seed)
 }
+
+func GenerateEvents(numEvents, fieldsPerLevel, depth int, withCollisions bool) []beat.Event {
```

Comment:
Great idea to extract it in a separate helper!

I think we could remove the collision support / benchmarks to streamline the interface. This use-case is very rare.